src/store.h

```diff
@@ -117,8 +115,13 @@ class MetadataStore {
   std::map<std::string, MonitoredResource> resource_map_;
   // A lock that guards access to the metadata map.
   mutable std::mutex metadata_mu_;
-  // A map from Full Resource Name to (JSON) resource metadata.
-  std::map<std::string, Metadata> metadata_map_;
+
+  // MetadataKey is a pair of the Full Resource Name,
+  // https://cloud.google.com/apis/design/resource_names#full_resource_name
+  // and the version that uniquely identifies a metadata blob in a cluster.
+  using MetadataKey = std::pair<std::string, std::string>;
+  // A map from the metadata key to (JSON) resource metadata.
+  std::map<MetadataKey, Metadata> metadata_map_;
```

Comment:
Just noting that we've discussed the possibility of a 2-level map (name -> version -> blob) and decided to do this later if necessary.